modules/exploits/windows/local/cve_2020_0787_bits_arbitrary_file_move.rb

```diff
@@ -0,0 +1,281 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = ExcellentRanking
+
+  include Msf::Post::Windows::Priv
+  include Msf::Exploit::EXE # Needed for generate_payload_dll
+  include Msf::Post::Windows::FileSystem
+  include Msf::Post::Windows::ReflectiveDLLInjection
+  include Msf::Exploit::FileDropper
+  include Msf::Post::File
+  include Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Background Intelligent Transfer Service Arbitrary File Move Privilege Elevation Vulnerability',
+        'Description' => %q{
+          This module exploits CVE-2020-0787, an arbitrary file move vulnerability in outdated versions of the
+          Background Intelligent Transfer Service (BITS), to overwrite C:\Windows\System32\WindowsCoreDeviceInfo.dll
+          with a malicious DLL containing the attacker's payload.
+
+          To achieve code execution as the SYSTEM user, the Update Session Orchestrator service is then started, which
+          will result in the malicious WindowsCoreDeviceInfo.dll being run with SYSTEM privileges due to a DLL hijacking
+          issue within the Update Session Orchestrator Service.
+
+          Note that presently this module only works on Windows 10 and Windows Server 2016 and later as the
+          Update Session Orchestrator Service was only introduced in Windows 10. Note that only Windows 10 has been tested,
+          so your mileage may vary on Windows Server 2016 and later.
+        },
+        'License' => MSF_LICENSE,
+        'Author' =>
+          [
+            'itm4n', # PoC
```

Comment:
@gwillcox-r7: I think @itm4n discovered this as well.

Reply:
https://itm4n.github.io/cve-2020-0787-windows-bits-eop/

Reply:
Yep he did, want me to update the comment on this to reflect that?